src/pybind/mgr/dashboard_v2/tox.ini

```diff
@@ -36,6 +36,8 @@ setenv=
 commands =
     bash ../src/vstart.sh -n
     sleep 10
+    python ./bin/ceph mgr module disable dashboard_v2
+    sleep 5
```

Comment:
why do you need to add a 5 second sleep after disabling the dashboard?

Reply:
because the CLI command returns before the module is fully deactivated.

Reply:
What about checking the mgr to check whether the module has been disabled or not? Also, any idea why the command returns before the module has been disabled? That does seem like it's weirdly breaking away the chains we usually put around user-facing commands.

Reply:
I'm going to be honest here. CI scripts never get the dedication they deserve, because there are always more important things to do.
I would be glad if someone could spend some time improving these scripts to avoid these shortcomings.

Reply:
@rjfd I don't know how to follow up your comment. Do you need someone to fix something so you address the `sleep 5`? Do you intend to address it, or maybe not? Do you need someone else to address something else? What is the shortcoming you are referring to?

Reply:
I meant that currently I don't have the time to improve the tox script, and since it's working as it is, I don't think it's a high priority.

Reply:
Interestingly, `ceph mgr module disable dashboard_v2 && ceph mgr module enable dashboard_v2` does work for me reliable. Did you had strange effects when skipping the sleep?

Reply:
@sebastian-philipp yes

Reply:
@rjfd what kind of problems did the absence of the sleep raise? AFAICT, that command should *never* return immediately (without having reflected the side-effects on the map, of course), and if it does it's likely a bug.

Reply:
without the sleeps I have seen that the `enable` sometimes fails and the module is not loaded. I haven't investigated the root cause.

Reply:
in essence, the sleep is required because the commands themselves (`ceph mgr module disable/enable <module>`) are talking to the monitors and changing the MgrMap. `enabling` or `disabling` is basically adding or removing a module from a list in the map. These changes will only be visible, later on, to the mgr, upon this new map epoch is received by the mgr itself.
Between the command returning and the mgr receiving a map, there is no actual time bound. It may take a few milliseconds, or it may take any amount of seconds. 5 seconds seems reasonable enough, simply because the mgr will likely have gotten the map in the mean time and done something about it.
@rjfd promised he would add some more info, in the form of a brief comment, to the patch, justifying the sleep for posteriry.

Reply:
The sleep is reasonable IMO. It can be improved by doing the sleep in a loop, which is broken out of when "it" (whatever is expected to happen) is found to have happened, but often a hard-coded sleep is enough to achieve a reasonable degree of reliability.